standalone-metastore/metastore-catalog/src/test/java/org/apache/iceberg/hive/Friend.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+/**
+ * Friend of CachedClientPool.
+ */
+public class Friend {
+  public static void cleanPoolCache() {
+      CachedClientPool.clientPoolCache().invalidateAll();
```

Comment:
Why is this class needed? A public static method calls another class's static method. 
The name of the class is weird.

Reply:
CacheClientPool.clientPoolCache() is package protected thus has to be called from a class in that package; this class mimics what a friend is in C++ parlance by residing in that expected package but making the call public. It is only meant for testing and that clean up is absolutely necessary when running multiple tests in sequence (Thanks Zhihua again :-)).

Reply:
why don't you move `HMSTestBase` to `org.apache.iceberg`? `Friend` class name is just meaningless

Reply:
I know you name is Zoltan :-) Zhihua found that this cache was the cause of not being able to run the tests in sequence (they worked 1 by 1).

Reply:
I renamed the class and provided a more meaningful explanation for its presence.